makefiles/boards/sam0.inc.mk

```diff
@@ -44,12 +44,10 @@ ifeq ($(PROGRAMMER),edbg)
   endif
 endif
 
-# this board uses J-Link for debug and possibly flashing
 ifeq ($(PROGRAMMER),jlink)
```

Comment:
In [#12654](https://github.com/RIOT-OS/RIOT/pull/12654/files#diff-8c5d501501e22a0cd41d19754359a66aR48) I used `DEBUG_ADAPTER` here. Not sure if this is sensible. It seemed to be, as it is more about the debugger than the programmer (also according to the comments).

Reply:
Will it still be used for flashing if I set `DEBUG_ADAPTER`?
Because that was the main reason for this PR - OpenOCD is rather unreliable with this device, but when using the vendor provided `JLinkExe` it works more reliable and faster.

Reply:
Mhhh...

https://github.com/RIOT-OS/RIOT/blob/756cab938538517e326701f5b1c13f86715e3554/makefiles/boards/sam0.inc.mk#L29-L36

this makes it difficult to say, only if `JLinkExe` is in `PATH` and only if the user does not provide any other `PROGRAMMER` via the environment. If kept as is it will most likely also lead to problems if `DEBUG_ADAPTER == jlink` but `PROGRAMMER != jlink`, but this is also the case in current master and this definitely solves the problem at hand in #12652.